Resync `html/anonymous-iframe` from WPT upstream

Ahmad-S792 opened the pull request:
#### d496804bab0a4678d7d59a23a44ac60ce8eb8f85
<pre>
Resync `html/anonymous-iframe` from WPT upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=297341">https://bugs.webkit.org/show_bug.cgi?id=297341</a>
<a href="https://rdar.apple.com/158239880">rdar://158239880</a>

Reviewed by Tim Nguyen.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/fa80f49be6ced958bce13b11c5d7bbc1c9a68d47">https://github.com/web-platform-tests/wpt/commit/fa80f49be6ced958bce13b11c5d7bbc1c9a68d47</a>

* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/anonymous-iframe-popup.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/anonymous-iframe-popup.tentative.https.window.js:
(promise_setup.async t.async const):
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/anonymous-window.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/anonymous-window.tentative.https.window.js:
(promise_test_parallel.async t):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/cache-storage.tentative.https.window.js:
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/cookie-store.tentative.https.window.js:
(promise_setup.async await):
(promise_setup):
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/cookie.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/cookie.tentative.https.window.js:
(const.cookieFromCredentiallessIframeRequest.async iframe_origin):
(promise_test_parallel.async test.promise_test_parallel.async test):
(promise_test_parallel.async test):
(const.cookieFromAnonymousIframeRequest.async iframe_origin): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/embedding.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/embedding.tentative.https.window.js:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/fenced-frame-bypass.tentative.https.window.js:
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/fenced-frame.tentative.https.window.js:
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/hasStorageAccess.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/hasStorageAccess.tentative.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/hasStorageAccess.tentative.https.window.js: Added.
(setup):
(const.hasStorageAccess):
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/indexeddb.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/indexeddb.tentative.https.window.js:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/initial-empty-document.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/initial-empty-document.tentative.https.window.js:
(promise_test_parallel.async t):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage-initial-empty-document.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage-initial-empty-document.tentative.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage-initial-empty-document.tentative.https.window.js: Added.
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage.tentative.https.window.js:
(const.store.async iframe):
(const.load.async iframe):
(promise_test.async test):
(promise_test_parallel.async test.async promise_test_parallel): Deleted.
(promise_test_parallel.async test): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccess.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccess.tentative.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccess.tentative.https.window.js: Added.
(setup):
(const.requestStorageAccess):
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccessFor.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccessFor.tentative.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccessFor.tentative.https.window.js: Added.
(setup):
(const.requestStorageAccessFor):
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/require-corp-embed-anonymous-iframe.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/require-corp-embed-anonymous-iframe.tentative.https.window.js:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/resources/common.js:
(const.newIframeCredentialless):
(const.newIframe):
(const.newFencedFrame.async child_origin):
(const.newAnonymousIframe): Deleted.
(const.newFencedFrame): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/resources/embedding-test.js:
(const.embeddingTest):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/serviceworker-partitioning.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/serviceworker-partitioning.tentative.https.window.js:
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/session-storage.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/session-storage.tentative.https.window.js:
(const.store.async iframe):
(const.load.async iframe):
(promise_test.async test):
(promise_test_parallel.async test.async promise_test_parallel): Deleted.
(promise_test_parallel.async test): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/sharedworker-partitioning.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/sharedworker-partitioning.tentative.https.window.js:
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/web-lock.tentative.https.window.js:
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/worker-cookies.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/worker-cookies.tentative.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/worker-cookies.tentative.https.window.js: Added.
(promise_test.async test):
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccess.tentative.https.window-expected.txt: Added.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/anonymous-iframe/require-corp-embed-anonymous-iframe.tentative.https.window-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage.tentative.https.window-expected.txt: Added.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccess.tentative.https.window-expected.txt: Added.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/anonymous-iframe/require-corp-embed-anonymous-iframe.tentative.https.window-expected.txt: Added.

Canonical link: <a href="https://commits.webkit.org/299417@main">https://commits.webkit.org/299417@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e16224d04582b1525ad681a81b42924f299e836e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118912 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38593 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29244 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125107 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70975 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c6206ade-58b1-4d8d-8270-aa057b5a74b2) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39289 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47175 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90241 "Found 1 new test failure: imported/w3c/web-platform-tests/encoding/encodeInto.any.serviceworker.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59756 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/da730092-c343-4e4b-b498-7eef73a08097) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121865 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31315 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106607 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70746 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9808939f-59af-463a-92fa-770c9fa284e9) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30378 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24721 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68764 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100758 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24909 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128152 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45819 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34611 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98900 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46185 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102827 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98680 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44133 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22136 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/42384 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18940 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45689 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45155 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48499 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46839 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
There's not much point resyncing this because it's not supported by WebKit, but I guess it's better than leaving outdated tests in the tree

Ahmad-S792 committed:
Resync `html/anonymous-iframe` from WPT upstream
https://bugs.webkit.org/show_bug.cgi?id=297341
rdar://158239880

Reviewed by Tim Nguyen.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/fa80f49be6ced958bce13b11c5d7bbc1c9a68d47

* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/anonymous-iframe-popup.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/anonymous-iframe-popup.tentative.https.window.js:
(promise_setup.async t.async const):
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/anonymous-window.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/anonymous-window.tentative.https.window.js:
(promise_test_parallel.async t):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/cache-storage.tentative.https.window.js:
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/cookie-store.tentative.https.window.js:
(promise_setup.async await):
(promise_setup):
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/cookie.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/cookie.tentative.https.window.js:
(const.cookieFromCredentiallessIframeRequest.async iframe_origin):
(promise_test_parallel.async test.promise_test_parallel.async test):
(promise_test_parallel.async test):
(const.cookieFromAnonymousIframeRequest.async iframe_origin): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/embedding.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/embedding.tentative.https.window.js:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/fenced-frame-bypass.tentative.https.window.js:
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/fenced-frame.tentative.https.window.js:
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/hasStorageAccess.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/hasStorageAccess.tentative.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/hasStorageAccess.tentative.https.window.js: Added.
(setup):
(const.hasStorageAccess):
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/indexeddb.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/indexeddb.tentative.https.window.js:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/initial-empty-document.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/initial-empty-document.tentative.https.window.js:
(promise_test_parallel.async t):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage-initial-empty-document.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage-initial-empty-document.tentative.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage-initial-empty-document.tentative.https.window.js: Added.
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage.tentative.https.window.js:
(const.store.async iframe):
(const.load.async iframe):
(promise_test.async test):
(promise_test_parallel.async test.async promise_test_parallel): Deleted.
(promise_test_parallel.async test): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccess.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccess.tentative.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccess.tentative.https.window.js: Added.
(setup):
(const.requestStorageAccess):
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccessFor.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccessFor.tentative.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccessFor.tentative.https.window.js: Added.
(setup):
(const.requestStorageAccessFor):
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/require-corp-embed-anonymous-iframe.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/require-corp-embed-anonymous-iframe.tentative.https.window.js:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/resources/common.js:
(const.newIframeCredentialless):
(const.newIframe):
(const.newFencedFrame.async child_origin):
(const.newAnonymousIframe): Deleted.
(const.newFencedFrame): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/resources/embedding-test.js:
(const.embeddingTest):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/serviceworker-partitioning.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/serviceworker-partitioning.tentative.https.window.js:
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/session-storage.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/session-storage.tentative.https.window.js:
(const.store.async iframe):
(const.load.async iframe):
(promise_test.async test):
(promise_test_parallel.async test.async promise_test_parallel): Deleted.
(promise_test_parallel.async test): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/sharedworker-partitioning.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/sharedworker-partitioning.tentative.https.window.js:
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/web-lock.tentative.https.window.js:
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/worker-cookies.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/worker-cookies.tentative.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/worker-cookies.tentative.https.window.js: Added.
(promise_test.async test):
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccess.tentative.https.window-expected.txt: Added.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/anonymous-iframe/require-corp-embed-anonymous-iframe.tentative.https.window-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/anonymous-iframe/local-storage.tentative.https.window-expected.txt: Added.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/anonymous-iframe/requestStorageAccess.tentative.https.window-expected.txt: Added.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/anonymous-iframe/require-corp-embed-anonymous-iframe.tentative.https.window-expected.txt: Added.

Canonical link: https://commits.webkit.org/299417@main
